hack/generator/pkg/codegen/pipeline/status_augment.go

```diff
@@ -63,32 +64,37 @@ func fuseAugmenters(augments ...augmenter) augmenter {
 
 // flattenAugmenter copies across the "flatten" property from Swagger
 func flattenAugmenter(allTypes astmodel.ReadonlyTypes) augmenter {
```

Comment:
I've read this a dozen times and I still have no idea what this method is doing. 

Strongly suggest extracting giving all those anonymous functions some intention revealing names.

Reply:
can you take another look after the big comment expansion I’ve done? the functions are highly contextual; in a language with proper pattern-matching they would distinct branches in a function; here they could be given names like `flattenAugment_handleTypeTypeName` and `flattenAugment_handleTypeNameType` but I don’t think that’s going to help much

Reply:
That's much more decipherable now! Thanks.